From 2bb57d3e938e300240672fca54b4e2f9f9615eff Mon Sep 17 00:00:00 2001
From: Janusz Dobrowolski <info@ngicon.com>
Date: Wed, 25 Jan 2023 22:03:12 +0100
Subject: [PATCH 5/5] Fixed error retrieving grade amount from empty table.

---
 includes/db/grade_db.inc | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/includes/db/grade_db.inc b/includes/db/grade_db.inc
index fb5e75e..3b9e068 100644
--- a/includes/db/grade_db.inc
+++ b/includes/db/grade_db.inc
@@ -33,9 +33,8 @@ function get_position_grades($position) {
 function get_grade_amount($position, $grade) {
 	$sql = "SELECT amount FROM ".TB_PREF."grade_table WHERE position_id = ".db_escape($position)." AND grade_id = ".db_escape($grade);
 	$result = db_query($sql, 'could not get grade data');
-	$amount = db_fetch($result)[0];
-
-	return $amount;
+	$row = db_fetch($result);
+	return $row ? $row[0]: 0;
 }
 
 function get_all_grades() {
-- 
2.25.1

